subjects/gaming/2.5d-adventure/README.md

```diff
@@ -0,0 +1,146 @@
+
+> Tip: For maximum learning, it is recommended to start this project with a blank Unreal project and implement everything from scratch.
+
+#### Main Menu
+
+The main menu `widget` should:
+
+- Be on a separate `level/map` from the `main game` map.
+- Contain two buttons:
+  - Start game button.
+  - Exit game button.
+
+#### Player Character
+
+The player `character` should:
+
+- Have a skeletal mesh.
+- Move only along two axes: left-right and up-down.
+- Have a basic locomotion system with the following animations:
+  - `Idle`.
+  - `Walking`.
+  - `Running`.
+  - `Jumping`.
+  - `Falling`.
+  - `Landing`.
+- Transition smoothly between a walking and running animations based on their speed.
+<center>
+<img src="./resources/locomotion.gif?raw=true" style = "width: 500px !important; height: 350px !important;"/>
+</center>
+
+> Tip: Look into the best practices for creating a locomotion system, Animation blueprints and Blend spaces.
+
+#### Collectible
+
+The collectible `Actor` should:
+
+- Have a static mesh.
+- Rotate around an axis of your choice.
+- Have a box collider that acts like a trigger.
+- Be collected when the player enters its box trigger.
+- Play a sound when collected
+<center>
+<img src="./resources/collectible.gif?raw=true" style = "width: 500px !important; height: 350px !important;"/>
+</center>
+
+#### HUD
+
+The HUD `widget` should:
+
+- Display a value related to the `collectible`.
+- Be updated each time a `collectible` is picked up.
+<center>
+<img src="./resources/hud.gif?raw=true" style = "width: 500px !important; height: 350px !important;"/>
+</center>
+
+#### Enemy
+
+The enemy `character` should:
+
+- Have a skeletal mesh.
+- Have at least a walking cycle.
+- Have a simple AI that patrols between two set points.
+- Kill the player on collision.
+- Be killed if the player lands on top of it.
+- Play a sound when killed.
+- create an instance of `Collectible` at the place of death.
+<center>
+  <div style="display: flex; justify-content: center;">
+    <img src="./resources/enemywalk.gif?raw=true" style="width: 500px !important; height: 350px !important;" />
+    <img src="./resources/enemydie.gif?raw=true" style="width: 500px !important; height: 350px !important;" />
+  </div>
+</center>
+
+#### Game Loop Logic
+
+The game loop consists of a spawn point where the player `starts` and `respawns` when dead.
+And an end point that defines the player goal to finish the level.
+When the player reaches the end point:
+
+- a menu with options to `restart` or `quit` the game should be displayed.
+
+#### Level design
+
+Your level design shouldn't consist of just a long, empty run to the finish point. However, since creating a level is a creative process, you have the freedom to design it as you see fit. For example, you can include moving platforms or skill-based platforming elements. The choice is yours to provide the player with a fun and challenging experience.
```

Comment:
as we ask this in the audit:
" Does the level include elements like moving platforms or skill-based platforming challenges?"

we should make it clear in this section of the readme that the student needs to implement some of these features

Reply:
i wanted to keep level design to be a creative process for the student hence not telling them exactly what elements to implement, should i rephrase it in the audit to make it so that they can implement whatever mechanics they want ? because thats what i was initially going for, and the moving platforms and skill based platforming were just examples.

Reply:
yes, you can do that as well.
I understood but then they would probably complain that we didn't ask specifically for that on the readme

Reply:
`Does the level include elements like moving platforms or skill-based platforming challenges?`

maybe this could be rephrased to :

`Does the level include mechanics chosen by the student?`

this will make it mandatory to implement some mechanic but not exactly specifying one.

or should i find another way to say it in the subject too?

what do you think?